lib/stimulus/engine.rb

```diff
@@ -4,6 +4,19 @@ module Stimulus
   class Engine < ::Rails::Engine
     config.autoload_once_paths = %w( #{root}/app/helpers )
 
+    if const_defined? :Sprockets
+      require "stimulus/directive_processor"
+
+      Sprockets.register_mime_type "application/importmap+json", extensions: ['.json.importmap']
+      Sprockets.register_preprocessor "application/importmap+json", DirectiveProcessor.new(comments: ["//", ["/*", "*/"]])
+      Sprockets.register_bundle_processor "application/importmap+json", Sprockets::Bundle
+      Sprockets.register_bundle_metadata_reducer "application/importmap+json", :data, proc { +"{}" } do |buffer, source|
+        buffer_json = ActiveSupport::JSON.decode(buffer)
+        source_json = ActiveSupport::JSON.decode(source)
+        ActiveSupport::JSON.encode(buffer_json.deep_merge(source_json))
+      end
+    end
+
     initializer "stimulus.assets" do
       Rails.application.config.assets.precompile += %w( importmap.json stimulus/manifest )
```

Comment:
Does it make sense to wrap this one into `if const_defined? :Sprockets` as well?